python/lancedb/table.py

```diff
@@ -534,7 +534,9 @@ def create(
     @classmethod
     def open(cls, db, name):
         tbl = cls(db, name)
-        if not os.path.exists(tbl._dataset_uri):
+        fs, path = pa.fs.FileSystem.from_uri(tbl._dataset_uri)
```

Comment:
Is this the only place we do this existence check? Should be extract a method and use it in all instances ?

Reply:
This is the only place `path.exists()` is called. In current API, yes.